paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt

```diff
@@ -613,6 +618,21 @@ internal class DefaultPaymentSheetLoader @Inject constructor(
         }
     }
 
+    private fun isCvcRecollectionRequired(
+        initializationMode: PaymentSheet.InitializationMode,
+        intent: StripeIntent
+    ): Boolean {
+        val deferredIntentMode = initializationMode as? PaymentSheet.InitializationMode.DeferredIntent
+        val deferredIntentRequiresCvcRecollection =
+            deferredIntentMode?.intentConfiguration?.requireCvcRecollection == true &&
+            deferredIntentMode.intentConfiguration.mode is PaymentSheet.IntentConfiguration.Mode.Payment
+
+        val paymentIntentRequiresCvcRecollection = (intent as? PaymentIntent)?.requireCvcRecollection == true &&
+            initializationMode !is PaymentSheet.InitializationMode.DeferredIntent
```

Comment:
Why is this necessary? Do we have a test for it?

Reply:
Deferred intents will appear as PaymentIntents, right? What else would they show up as?

Reply:
We wouldn't have the intent at this point if it's deferred, no?

Reply:
Added back and added a test.